tests/fault_tolerance/deploy/test_deployment.py

```diff
@@ -147,7 +147,18 @@ async def test_fault_scenario(
         scenario.deployment.set_model(scenario.model)
         model = scenario.model
     else:
-        model = scenario.deployment["VllmDecodeWorker"].model
+        # Get model from the appropriate worker based on backend
+        try:
+            if scenario.backend == "vllm":
+                model = scenario.deployment["VllmDecodeWorker"].model
+            elif scenario.backend == "sglang":
+                model = scenario.deployment["decode"].model
```

Comment:
NOTE (not for this PR) - we should revisit consistency of worker/component names in the 3 backends. `VllmDecodeWorker` for vllm vs `decode` for sglang. CC @ishandhanani